azurerm/resource_arm_storage_account.go

```diff
@@ -1484,6 +1568,27 @@ func flattenStorageAccountVirtualNetworks(input *[]storage.VirtualNetworkRule) [
 	return virtualNetworks
 }
 
+func flattenBlobProperties(input storage.BlobServiceProperties) []interface{} {
+	if input.BlobServicePropertiesProperties == nil {
+		return []interface{}{}
+	}
+
+	blobProperties := make(map[string]interface{})
+
+	if deletePolicy := input.BlobServicePropertiesProperties.DeleteRetentionPolicy; deletePolicy != nil {
+		if enabled := deletePolicy.Enabled; enabled != nil && *enabled {
+			deleteRetentionPolicy := make([]interface{}, 0)
+			attr := make(map[string]interface{})
+
+			attr["days"] = int(*deletePolicy.Days)
```

Comment:
Done, but I've got one question to this one. 

In the end `days == 0` when the Azure API returns that the `DeleteRetentionPolicy` is `enabled` but the number of days are missing. This should never happen I guess, but `0` is not a valid value for `days`. This will lead to an error I guess. Shouldn't we just throw an error with appropriate message in this case?

Reply:
The error wouldn't be something the user can do much about unfortunately - in this instance we're setting this to `0` since there's a nested object/list here, and Terraform 0.12 requires that all keys are set in nested elements - so this is mostly to keep Terraform Core happy more than anything